flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/expressions/ScalarOperatorsTest.scala

```diff
@@ -228,6 +228,15 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
     testSqlApi("uuid() = cast(f22 as timestamp_ltz)", "NULL")
   }
 
+  @Test
+  def testTimePointTypeNotEqualsStringType(): Unit = {
+    testSqlApi("f22 = '1996-11-10 12:34:56'", "TRUE")
```

Comment:
I see that the datetime family contains 5 datatypes(see `org.apache.flink.table.types.logical.LogicalTypeRoot`), so it is recommended to add more tests for complete coverage of the datatypes